egs2/TEMPLATE/svs1/README.md

```diff
@@ -0,0 +1,540 @@
+    * [8\. Objective evaluation](#8-objective-evaluation)
+    * [9\. Model packing](#9-model-packing)
+  * [How to run](#how-to-run)
+    * [Naive_RNN training](#naive_rnn-training)
+    * [XiaoiceSing training](#xiaoicesing-training)
+    * [VISinger training](#visinger-training)
+    * [Multi speaker model with speaker ID embedding training](#multi-speaker-model-with-speaker-id-embedding-training)
+    * [Multi language model with language ID embedding training](#multi-language-model-with-language-id-embedding-training)
+    * [Vocoder training](#vocoder-training)
+    * [Evaluation](#evaluation)
+  * [About data directory](#about-data-directory)
+  * [Supported text frontend](#supported-text-frontend)
+  * [Supported text cleaner](#supported-text-cleaner)
+  * [Supported Models](#supported-models)
+
+
+## Recipe flow
+
+SVS recipe consists of 8 stages.
+
+### 1. Database-dependent data preparation
+
+Data preparation stage.
+
+It calls `local/data.sh` to creates Kaldi-style data directories but with additional `midi.scp` and `label` in `data/` for training, validation, and evaluation sets.
+
+See also:
+- [About data directory](#about-data-directory)
+
+### 2. Wav dump / Embedding preparation
+
+If you specify `--feats_type raw` option, this is a wav dumping stage which reformats `wav.scp` in data directories.
+
+Else, if you specify `--feats_type fbank` option or `--feats_type stft` option, this is a feature extracting stage (to be updated).
+
+MIDI is also normalized and segmented at this stage.
+
+Additionally, speaker ID embedding and language ID embedding preparation will be performed in this stage if you specify `--use_sid true` and `--use_lid true` options.
+Note that this processing assume that `utt2spk` or `utt2lang` are correctly created in stage 1, please be careful.
+
+### 3. Filtering (Removal of long / short data)
+
+Filtering stage.
+
+Processing stage to remove long and short utterances from the training and validation sets. 
+You can change the threshold values via `--min_wav_duration` and `--max_wav_duration`.
+
+Empty text will also be removed.
+
+### 4. Token list generation
+
+Token list generation stage.
+It generates token list (dictionary) from `srctexts`.
+You can change the tokenization type via `--token_type` option. `token_type=phn` are supported.
+If `--cleaner` option is specified, the input text will be cleaned with the specified cleaner.
+If `token_type=phn`, the input text will be converted with G2P module specified by `--g2p` option.
+
+See also:
+- [Supported text cleaner](#supported-text-cleaner).
+- [Supported text frontend](#supported-text-frontend).
+
+Data preparation will end in stage 4. You can skip data preparation (stage 1 ~ stage 4) via `--skip_data_prep` option.
+
+### 5. SVS Statistics collection
+
+Statistics calculation stage.
+It collects the shape information of the input and output and calculates statistics for feature normalization (mean and variance over training and validation sets).
+
```

Comment:
store pitch (and feats? to be discussed) in stage 5 via `--write_collected_feats true`